### DISCUSSION TITLE
Using @Shared in tests

### DISCUSSION THREAD

**Discussion by kgrigsby59** [pointfreeco/swift-composable-architecture#3330] [General]
I have a simple test using a shared UserDefault that is failing and I don't understand why. The shared value in the Feature state is being set to the default value (true) instead of the value (false) set in the test. I'm using Xcode 16.0 beta 6 and tried both Swift 5 and Swift 6.
```
extension PersistenceKey where Self == PersistenceKeyDefault<AppStorageKey<Bool>> {
    public static var boolValue: Self {
        PersistenceKeyDefault(.appStorage("boolValue"), true)
    }
}

@Reducer
struct Feature {
    struct State: Equatable {
        @Shared(.boolValue) var boolValue
    }
}
```
```
import ComposableArchitecture
import Testing
@testable import ShareTest

@MainActor
struct ShareTestTests {

    @Test func example() {
        @Shared(.boolValue) var boolValue = false

        let store = TestStoreOf<Feature>(initialState: .init()) {
            Feature()
        }
        #expect(boolValue == store.state.boolValue)
        print("boolValue: \(boolValue)") // false
        print("feature.boolValue: \(store.state.boolValue)") // true - why?
    }

}
```
[ShareTest.zip](https://github.com/user-attachments/files/16819879/ShareTest.zip)

**Comment by kgrigsby59** [comment]
From @mbrandonw in Slack:

> This is unfortunately a problem with how we are trying to support Swift Testing in TCA. What we really need is the CustomExecutionTrait from the library, but they have it currently hidden behind @_spi. And so we are trying to emulate it, but it isn't perfect.
> we will update docs, but the rule is that your TestStore should basically be the very first thing done in a test function, and then you can interact with shared if you need to set it up
>

**Reply by yuanhang** [reply]
Hi @kgrigsby59, would you please share the workaround? I tried to create the TestStore first as you referenced, but I still got the same failed test.
